src/api/config.c

```diff
@@ -1118,3 +1118,49 @@ int api_config(struct ftl_conn *api)
 
 	return 0;
 }
+
+int api_config_properties(struct ftl_conn *api)
+{
+	cJSON *read_only = JSON_NEW_ARRAY();
+
+	// Iterate over all known config elements
+	for(unsigned int i = 0; i < CONFIG_ELEMENTS; i++)
+	{
+		// Get pointer to memory location of this conf_item
+		struct conf_item *conf_item = get_conf_item(&config, i);
+
+		const char *reason = NULL;
+		const char *description = NULL;
+		if(conf_item->f & FLAG_ENV_VAR)
+		{
+			reason = "env_var";
+			description = "Set via environment variable";
+		}
+		else if(config.misc.readOnly.v.b)
+		{
+			reason = "read_only";
+			description = "Config is in read-only mode";
+		}
+		else if(conf_item->f & FLAG_READ_ONLY)
```

Comment:
Is this case still needed after you added the case for `misc.readOnly`. Because I think `FLAG_READ_ONLY` is only set once `misc.readOnly` is true.